front-end/src/visualization/Visualization.jsx

```diff
@@ -50,47 +57,115 @@ function Visualization() {
+                }
+
+                const data = await response.json();
+                setTitle(data.title || 'Untitled');
+            } catch (error) {
+                console.error('Error fetching stitch name:', error);
+            }
+        };
+
+        fetchStitchName();
+        getClusterData();
+    }, []);
+
+    useEffect(() => {
+        if (clusterData.centroids.length > 0) {
+            const clusterPositions = calculateClusterPositions(clusterData.centroids.length, 1000, 1000, 250);
+            const { nodes, edges } = generateClusterData(clusterData, clusterPositions);
+            setGraphData({ nodes, edges });
         }
+    }, [clusterData]);
+
+    const handleToStitch = () => {
+        navigate(`/${username}/create`, { state: { stitchId } });
     }
 
-    const clusterPositions = calculateClusterPositions(3, 1000, 1000, 200);
-    let globalId = 1;
-    let allNodes = [];
-    let allEdges = [];
-    clusterPositions.forEach((pos, index) => {
-        const clusterData = generateClusterData(globalId++, `Cluster ${index + 1}`, pos.x, pos.y, 5, 70);
-        allNodes = allNodes.concat(clusterData.nodes);
-        allEdges = allEdges.concat(clusterData.edges);
-    });
-    addInterClusterEdges(allNodes, allEdges);
+
     return (
         <div className='Visualization'>
             <header>
                 <Navbar username={username} page={"visualization"} />
+                <Flex justifyContent="center" mt="1em">
+                    <Heading size='2xl'>{title}</Heading>
+                </Flex>
+                <Flex justifyContent="center">
+                    <Button
+                        className="navigateToVisualization"
+                        bgGradient="linear(to-r, rgba(115, 41, 123, 0.9), rgb(83, 41, 140, 0.9))"
+                        color="white"
+                        width="11em"
+                        mt="1em"
+                        _focus={{ boxShadow: 'none', bg: 'white', color: 'black' }}
+                        _active={{ boxShadow: 'none' }}
+                        _hover={{
+                            opacity: 1,
+                            backgroundSize: 'auto',
+                            boxShadow: '0 0 20px -2px rgba(195, 111, 199, .5)',
+                            transform: 'translate3d(0, -0.5px, 0) scale(1.01)',
+                        }}
+                        onClick={handleToStitch}
+                    >
+                        <Text fontSize="lg">Back to Stitch</Text>
+                    </Button>
+                </Flex>
+                <Flex justifyContent='center'>
+                    <Text color='white' fontSize='md' mt='1.5em' textAlign='center' width='70%'>According to Spotify, a value on valence that's closer to 1.0 will describe happy, cheerful or euphoric songs 🥳😺💃, while a value closer to 0.0 will relate sad, depressed or angry songs 😡😿💔. This visualization clusters songs with their closest valence neighbors!</Text>
```

Comment:
It would also be nice (not really in scope of your project timeline) to choose colors based on the meaning of the valence numbers instead of showing random colors- it helps user chunk up information better

Reply:
Just added that in the new pull request haha!

Reply:
> It would also be nice (not really in scope of your project timeline) to choose colors based on the meaning of the valence numbers instead of showing random colors- it helps user chunk up information better

Just added that in the new pull request!